### ISSUE TITLE
Incorrect Total Gain/Loss within Stock widget on the Dashboard

### ISSUE THREAD

**Issue by vbojan** [moneymanagerex/moneymanagerex#5969]
MMEX version:
 - [x] 1.6.0
 - [x] 1.6.1
 - [x] 1.6.2
 - [x] 1.6.3
 - [x] 1.6.4 Beta 6

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

The Stock widget on the main Dashboard shows Total amount under Gain/Loss column.
This should be a total of all gains and/or losses across all stock accounts.
Please look at the screenshot:

![image](https://github.com/moneymanagerex/moneymanagerex/assets/2696314/804346ea-8a9a-4ce1-9768-250bd0e41eb3)

This also exposed 2 tiny formatting issues related to USD currency:
1. `GROUP_SEPARATOR` field within `CURRENCYFORMATS_V1` table contains space instead of comma (as in older versions), and is missing values for `UNIT_NAME` and `CENT_NAME` (not sure if the last two are actually used, but some other currencies have them setup, and these used to be "dollar" and "cent").
2. Negative USD values should always place the negative symbol before the currency and amount.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:
1. Create new database.
2. Create new Investment account.
3. Create new stock investment: a) enter stock symbol, b) enter share number, c) enter share price, d) hit Save and then Close - this will crash MMEx (I was able to reproduce in 1.6.3, but didn't test older versions).
4. Open MMEx and confirm you'd like to continue to open this database.
5. Observe Stock widget at the dashboard - the initial Gain/Loss for created account will be $0.00.
6. Download Currency and Stock rates - this will now update all amounts, but Total under Gain/Loss column will still reflect whatever the Total amount is under the Total column.

Expected result:
Total row should present sums of relevant columns it belongs to, so in this case Total should be a sum of all Gains and/or Losses for all stock accounts.

Actual result:
Total amount for Gains/Losses contains the same value as Total amount across all of stock accounts.

**Comment by n-stein** [comment]
I think this is only an issue when there are no share transactions on the stocks, i.e. you add a stock, set a current value, but never actually make the initial purchase transaction.

**Comment by n-stein** [comment]
USD grouping char has been space since at least May 2018 per the history of tables.sql in the database submodule.

**Comment by n-stein** [comment]
PR #5972 opened to update the database submodule

**Comment by tactilis** [comment]
> USD grouping char was briefly **,** from May 2021 to Sept 2022. Not sure why it was reverted to space.

I don't remember any discussion about that change.

<br>

Microsoft say that the convention in the US is use a comma:
https://learn.microsoft.com/en-us/globalization/locale/number-formatting#the-character-used-as-the-thousands-separator

Wikipedia says comma but also suggests space (in education):
https://en.wikipedia.org/wiki/Decimal_separator#Examples_of_use

**Comment by n-stein** [comment]
> I don't remember any discussion about that change.

I think the reversal may have been accidental. It was updated in 
https://github.com/moneymanagerex/database/commit/35959091dd304d9ed8f29260f787a536ab6a7612 then reversed in the v15 update https://github.com/moneymanagerex/database/commit/2ae8a5a01c0ddafc1ba5fbd39e67bd38ddf5fc9b which significantly changed the initial CURRENCYFORMATS_V1 table.

I think it makes sense to use a comma in line with the `en_US` locale.

**Comment by n-stein** [comment]
> However, the crash happens even if share account is created and a corresponding transaction.

I don't get a crash on 1.6.3 or 1.6.4-Beta 6 following the steps you outlined. We did fix some crash issues related to stocks in 1.6.4 already. Would you please test the beta to see if you can reproduce your crash issue?
Find it here: https://ci.appveyor.com/project/whalley/moneymanagerex

> I just noticed something awkward I didn't pay attention before:
> "Some account" is the investment account whereas "Some company" is the optional name of the stock (i.e. AAPL if omitted for Apple), but when either is selected they show the same view with stock transactions for that investment account.
> In addition, right clicking on stock opens up a context menu for related investment account so attempt to "edit" "Some company" will actually bring a dialog to edit "Some account".
> This could be confusing to users.

"test" is the investment account. "Some account" is not the investment account, it is the share account for "Some Company". Clicking on "Some Company" opens the share account. Note that the system expects these to have the same name which makes this less confusing. I do think there is a minor issue here where the share account is displayed in "Favorites" even if "Display Share Accounts" is turned off. I'll open a new issue for this.

![image](https://github.com/moneymanagerex/moneymanagerex/assets/16599929/fd2fda8a-1223-4552-bfd5-d45849e8627a)
![image](https://github.com/moneymanagerex/moneymanagerex/assets/16599929/49abe999-97ff-4926-8589-d21548ecc10b)

> Speaking of US currency, I just noticed en-GB spelling of "Favo**u**rites" being used for en-US. It should be `Favorites`.

This is determined by your selected UI language not your locale. I suspect you selected "English (United Kingdom)"
![image](https://github.com/moneymanagerex/moneymanagerex/assets/16599929/e601cd26-162a-4f38-b367-f896b96171fa)

**Comment by n-stein** [comment]
> The value for `TOACCOUNTID` is then causing an unusual "<" prefix to Payee and Status

This is due to the way share transactions are currently recorded in MMEX. Although share transactions are stored as "Deposit" and "Withdrawal" they have a `TOACCOUNTID` as if they were a transfer. This is used to include/exclude them from income and expense in reporting

![image](https://github.com/moneymanagerex/moneymanagerex/assets/16599929/227d6337-c1c6-4911-8a8f-17b3d5c4635e)

For "Withdrawal" (a.k.a "Buy") transactions:
- Selecting "Transfer" records them as a Withdrawal from the source account to a dummy account "32702". These are excluded from income/expense calculations in reporting.
- Deselecting "Transfer" records them as a Withdrawal from the source account to dummy account "32701". These are counted as expenses in reporting.

For "Deposit" (a.k.a "Sell") transactions:
- Selecting "Transfer" records them as a Deposit in the target account coming from dummy account "32702". These are excluded from income/expense calculations in reporting.
- Deselecting "Transfer" records them as a Deposit in the target account coming from dummy account "32701". These are counted as income in reporting.

The `<` symbol is shown if you check the "Transfer" box for the share transaction and is how MMEX normally displays transfers. Normally the arrow points from the source account to the target account. This doesn't work very well for share transfers since they are really deposits/withdrawals within a single account, so they are displayed using the payee instead prefixed with a `<`. Ideally the arrow would switch directions for "Withdrawal" share transactions to follow the standard.

**Comment by vbojan** [comment]
I just tested the latest build of 1.6.4-Beta.6 (https://github.com/moneymanagerex/moneymanagerex/commit/6054bb870523783603d4b95284577e43a174bf78) and it doesn't crash anymore 💯 regardless if I create just the investment account or also create a share account and corresponding transaction, so everything looks good now.

The build that was crashing is from yesterday (d9e093a) and official 1.6.3 release is affected by the same/similar issue as it crashes under the same scenarios and I'm glad this is fixed now.

My MMEx was configured to use "System default", but even after switching to "English (United States)" it was showing "Favourites" and after looking closely I found that somehow I deleted en_US localization folder so it was defaulting to en_GB which is what appears to be hard-coded as that's what I see if I delete all *.mo files from the bin folder. Sorry about raising false alarm.

**Thank you so much** for explaining in depth the logic behind shares and their transactions. I had no idea it's per design and just felt awkward to me. It does have sense now and I appreciate you spending time to elaborate it.

**Comment by n-stein** [comment]
> I had no idea it's per design and just felt awkward to me.

It's not just you. Stocks/shares as they currently are implemented in MMEX are confusing and awkward to use. They were shoehorned into the old MMEX v7 database and need a rework to simplify and streamline common transactions (buy, sell, dividend, split, etc).
